File: CivitasRuby/pkg/CivitasRuby-0.0.1/lib/casilla.rb
#encoding:utf-8

require_relative 'jugador'
require_relative 'sorpresa'
require_relative 'mazo_sorpresas'
require_relative 'titulo_propiedad'


module Civitas
  class Casilla
    attr_reader :nombre
    attr_reader :titulo_propiedad
    
    def init
      @nombre=nil
      @titulo_propiedad = nil
      @importe=nil
      @carcel=nil
      @sorpresa=nil
      @mazo=nil
    end
    
    def initialize(nombre)
      init
      @nombre=nombre
    end
   
    def informe(actual,todos)
      Diario.instance.ocurre_evento("El jugador " + todos[actual].get_nombre + " ha caido en la casilla: " + @nombre)
    end
    
    def recibe_jugador(actual, todos)
      informe(actual, todos)
    end

    def jugador_correcto(actual,todos)
      bool=false
      if(actual<todos.length && actual >= 0)
        bool=true
      end
      return bool
    end

    def to_string
      String::new(str="**************************************************")
      str+="\n " + @nombre + " ‹(ᵒᴥᵒ )›"
      str+="\n**************************************************\n"
      
      return str
    end
  end
end

File: CivitasJava/src/civitas/Casilla.java
package civitas;

import java.util.ArrayList;


public class Casilla {
    private String nombre;
    int carcel;
    float importe;
    Sorpresa sorpresa;
    TituloPropiedad tituloPropiedad;
    MazoSorpresas mazo;
    
    Casilla(String nombre){
        init();
        this.nombre=nombre;  
    }

    public String getNombre(){
        return nombre;
    }
    
    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos){
        boolean bool = false;
        if(actual>=0 && actual<todos.size()){
            bool=true;
        }
        return bool;
    }

    public String toString(){

        String str="******************************";
        str+="\n " + getNombre() + " ‹(ᵒᴥᵒ )›";
        str+="\n******************************\n";
    
        return str;
    }
    
    TituloPropiedad getTituloPropiedad(){
        return this.tituloPropiedad;
    }
    
    public void recibeJugador(int actual, ArrayList<Jugador> todos){
        informe(actual, todos);
    }
    
    void informe(int actual, ArrayList<Jugador> todos){
        Diario.getInstance().ocurreEvento("El jugador " + todos.get(actual).getNombre() + " ha caido en la casilla: " + this.getNombre());
                
    }

    void init(){
        nombre=null;
        tituloPropiedad=null;
        importe=0;
        carcel=0;
        sorpresa=null;
        mazo=null;
    }
}
